R/zchunk_LA105.an_FAO_R_C_Y.R

```diff
@@ -38,100 +36,160 @@ module_aglu_LA105.an_FAO_R_C_Y_DISABLED <- function(command, ...) {
-    AGLU_ctry <- get_data(all_data, "aglu/AGLU_ctry")
     FAO_an_items_cal_SUA <- get_data(all_data, "aglu/FAO/FAO_an_items_cal_SUA")
-    FAO_an_items_PRODSTAT <- get_data(all_data, "aglu/FAO/FAO_an_items_PRODSTAT")
     L100.FAO_an_Food_t <- get_data(all_data, "L100.FAO_an_Food_t")
     L100.FAO_an_Prod_t <- get_data(all_data, "L100.FAO_an_Prod_t")
     L100.FAO_an_Stocks <- get_data(all_data, "L100.FAO_an_Stocks")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses translate_to_full_table
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Process FAO animal products food consumption data: map in GCAM region and commodities, convert units, aggregate to region and commodity
+    L100.FAO_an_Food_t %>%
+      left_join_error_no_match(iso_GCAM_regID, by = "iso")  %>%                               # Map in GCAM regions
+      left_join(FAO_an_items_cal_SUA, by = "item")  %>%                                       # Map in GCAM commodities
+      filter(!is.na(GCAM_commodity)) %>%                                                      # Remove commodities not included in GCAM
+      mutate(Mt = value * CONV_TON_MEGATON, Pcal = value * Mcal_t * CONV_MCAL_PCAL)  %>%      # Convert from tons to Mt and Pcal
+      group_by(GCAM_region_ID, GCAM_commodity, year) %>%                                      # Group by region, commodity, year
+      summarize(Mt = sum(Mt), Pcal = sum(Pcal)) %>%                                           # Aggregate food consumption in Mt and Pcal
+      ungroup() %>%                                                                           # Ungroup before complete
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),                        # Fill in missing region/commodity combinations with 0
+               GCAM_commodity, year, fill = list(Mt = 0, Pcal = 0)) %>%
+      # Calculate average caloric content of consumed commodities (kcal/g) for each region and animal type, set NA values to 1
+      mutate(value = if_else(Mt == 0, 1, Pcal / Mt))  ->
```

Comment:
This is a little confusing: the comment above said units are kcal/g, but this is diving 1e15 cal / 1e12 g = ...units work out, but might clarify in comment.